<center> <img src = https://raw.githubusercontent.com/AndreyRysistov/DatasetsForPandas/main/hh%20label.jpg alt="drawing" style="width:400px;">

# <center> Проект: Анализ вакансий из HeadHunter
   

In [279]:
import pandas as pd
import psycopg2
import warnings

warnings.filterwarnings('ignore')

In [280]:
# вставьте сюда параметры подключения из юнита 1. Работа с базой данных из Python 


In [281]:
connection = psycopg2.connect(
    dbname=DBNAME,
    user=USER,
    host=HOST,
    password=PASSWORD,
    port=PORT
)

# Юнит 3. Предварительный анализ данных

1. Напишите запрос, который посчитает количество вакансий в нашей базе (вакансии находятся в таблице vacancies). 

In [282]:
# текст запроса
query_3_1 = f'''select count(*) "количество вакансий в базе:"
from public.vacancies
'''
           

In [283]:
# результат запроса
df_3_1 = pd.read_sql_query(query_3_1, connection)

df_3_1

количество вакансий в базе:
0                        49197

2. Напишите запрос, который посчитает количество работодателей (таблица employers). 

In [284]:
# текст запроса
query_3_2 = f'''select count(*) "количество работодателей в базе:"
from public.employers
'''

In [285]:
# результат запроса
df_3_2 = pd.read_sql_query(query_3_2, connection)
df_3_2

количество работодателей в базе:
0                             23501

3. Посчитате с помощью запроса количество регионов (таблица areas).

In [286]:
# текст запроса
query_3_3 = f'''select count(*) "количество регионов:"
from public.areas
'''

In [287]:
# результат запроса
df_3_3 = pd.read_sql_query(query_3_3, connection)
df_3_3

количество регионов:
0                  1362

4. Посчитате с помощью запроса количество сфер деятельности в базе (таблица industries).

In [288]:
# текст запроса
query_3_4 = f'''select count(*) "количество сфер деятельности:"
from public.industries
'''

In [289]:
# результат запроса
df_3_4 = pd.read_sql_query(query_3_4, connection)
df_3_4

количество сфер деятельности:
0                            294

***

*Выводы:*
- По данным предварительного анализа данных можно увидеть в базе 294 сфер деятельности, в которых всего насчитывается 49197 вакансий от 23501 работодателей из различных регионов, точнее 1362 региона.

# Юнит 4. Детальный анализ вакансий

1. Напишите запрос, который позволит узнать, сколько (cnt) вакансий в каждом регионе (area).
Отсортируйте по количеству вакансий в порядке убывания.

In [290]:
# вычисляем количество вакансий в каждом регионе
query_4_1 = f'''select a.name area, count(v.*) cnt  
from public.vacancies v
join public.areas a on v.area_id = a.id
group by a.id
order by 2 desc
'''

In [291]:
# результат запроса
df_4_1 = pd.read_sql_query(query_4_1, connection)
df_4_1

area   cnt
0                      Москва  5333
1             Санкт-Петербург  2851
2                       Минск  2112
3                 Новосибирск  2006
4                      Алматы  1892
..                        ...   ...
764                    Кизляр     1
765                    Джизак     1
766                    Эртиль     1
767                  Арсеньев     1
768  Тура (Красноярский край)     1

[769 rows x 2 columns]

2. Напишите запрос, чтобы определить у какого количества вакансий заполнено хотя бы одно из двух полей с зарплатой.

In [292]:
# определяем количество вакансий, где заполнено хотя бы одно поле с з/п
query_4_2 = f'''select count(v.*) cnt_vacancies
from public.vacancies v
where v.salary_from is not null or v.salary_to is not null
'''

In [293]:
# результат запроса
df_4_2 = pd.read_sql_query(query_4_2, connection)
df_4_2

cnt_vacancies
0          24073

3. Найдите средние значения для нижней и верхней границы зарплатной вилки. Округлите значения до целого.

In [294]:
# запрос на определение среднего для пнижней и верхней границы з/п
query_4_3 = f'''select round(avg(salary_from)) avg_sal_from,
round(avg(salary_to)) avg_sal_to
from public.vacancies
'''

In [295]:
# результат запроса
df_4_3 = pd.read_sql_query(query_4_3, connection)
df_4_3

avg_sal_from  avg_sal_to
0       71065.0    110537.0

4. Напишите запрос, который выведет количество вакансий для каждого сочетания типа рабочего графика (schedule) и типа трудоустройства (employment), используемого в вакансиях. Результат отсортируйте по убыванию количества.


In [300]:
# выведение количества вакансий для сочетаний типа графика и типа трудоустройства
query_4_4 = f'''select schedule, employment, count(name)
from public.vacancies  
group by 1, 2
order by 3 desc
'''

In [301]:
df_4_4 = pd.read_sql_query(query_4_4, connection)
df_4_4

schedule           employment  count
0        Полный день     Полная занятость  35367
1   Удаленная работа     Полная занятость   7802
2      Гибкий график     Полная занятость   1593
3   Удаленная работа  Частичная занятость   1312
4     Сменный график     Полная занятость    940
5        Полный день           Стажировка    569
6     Вахтовый метод     Полная занятость    367
7        Полный день  Частичная занятость    347
8      Гибкий график  Частичная занятость    312
9        Полный день     Проектная работа    141
10  Удаленная работа     Проектная работа    133
11     Гибкий график           Стажировка    116
12    Сменный график  Частичная занятость    101
13  Удаленная работа           Стажировка     64
14     Гибкий график     Проектная работа     18
15    Сменный график           Стажировка     12
16    Вахтовый метод     Проектная работа      2
17    Сменный график     Проектная работа      1

5. Напишите запрос, выводящий значения поля Требуемый опыт работы (experience) в порядке возрастания количества вакансий, в которых указан данный вариант опыта. 

In [302]:
# выводим количество вакансий для поля Требуемый опыт работы
query_4_5 = f'''select experience, count(name)
from public.vacancies
group by experience
order by 2'''

In [303]:
# результат запроса
df_4_5 = pd.read_sql_query(query_4_5, connection)
df_4_5

experience  count
0         Более 6 лет   1337
1           Нет опыта   7197
2       От 3 до 6 лет  14511
3  От 1 года до 3 лет  26152

***

*Выводы:*
- по детальному анализу вакансий можно увидеть, что количество вакансий больше всего в регионах-миллионниках 
- в половине вакансий указывается хотя бы одно из полей с зарплатой
- в среднем в вакансиях предлагаются зарплаты от 71065 руб до 110537руб
- наиболее требуемое сочетание полной занятости с графиком полный день, второе по популярности полная занятость с удаленной работой
- наиболее требуемый опыт работы от 1 года до 3 лет.

# Юнит 5. Анализ работодателей

1. Напишите запрос, который позволит узнать, какие работодатели находятся на первом и пятом месте по количеству вакансий.

In [304]:
# выводим работодателя на первом месте по количеству вакансий
query_5_1 = f'''select e.name employer_№1, count(v.*) cnt_vac
from public.employers e
join public.vacancies v on e.id = v.employer_id
group by e.id
order by 2 desc
limit 1
'''

# выводим работодателя на пятом месте по количеству вакансий
query_5_1_1 = f'''select e.name employer_№5, count(v.*) cnt_vac
from public.employers e
join public.vacancies v on e.id = v.employer_id
group by e.id
order by 2 desc
offset 4 limit 1
'''

In [305]:
# результат первого места
df_5_1 = pd.read_sql_query(query_5_1, connection)
df_5_1

employer_№1  cnt_vac
0      Яндекс     1933

In [306]:
# результат пятого места
df_5_1_1 = pd.read_sql_query(query_5_1_1, connection)
df_5_1_1

employer_№5  cnt_vac
0  Газпром нефть      331

2. Напишите запрос, который для каждого региона выведет количество работодателей и вакансий в нём.
Среди регионов, в которых нет вакансий, найдите тот, в котором наибольшее количество работодателей.


In [307]:
# выводим количество работодателей для каждого региона по убыванию, среди регионов без вакансий
query_5_2 = f'''select a.name area, count(e.*) cnt_emp, count(v.*) cnt_vac
from public.areas a
left join public.employers e on a.id = e.area
left join public.vacancies v on a.id = v.area_id
where v.id is null
group by a.id
order by 2 desc
'''

In [308]:
# результат запроса
df_5_2 = pd.read_sql_query(query_5_2, connection)
df_5_2

area  cnt_emp  cnt_vac
0                Россия      410        0
1             Казахстан      207        0
2    Московская область       75        0
3    Краснодарский край       19        0
4    Ростовская область       18        0
..                  ...      ...      ...
588             Скидель        0        0
589          Белоозерск        0        0
590           Орловский        0        0
591          Павловская        0        0
592             Таксимо        0        0

[593 rows x 3 columns]

3. Для каждого работодателя посчитайте количество регионов, в которых он публикует свои вакансии. Отсортируйте результат по убыванию количества.


In [309]:
# выводим работодателей с количеством регионов, где публикует вакансии свои в порядке убывания 
query_5_3 = f'''select e.name, count(distinct v.area_id) cnt_area
from public.employers e
join public.vacancies v on e.id = v.employer_id
group by e.name
order by 2 desc
'''

In [310]:
# результат запроса
df_5_3 = pd.read_sql_query(query_5_3, connection)
df_5_3

name  cnt_area
0                      Яндекс       181
1                  Ростелеком       152
2                  Спецремонт       116
3      Поляков Денис Иванович        88
4                   ООО ЕФИН         71
...                       ...       ...
14761                  UniSol         1
14762            UNISTORY LLC         1
14763                   UNIT6         1
14764     United Distribution         1
14765              UNITEL LLC         1

[14766 rows x 2 columns]

4. Напишите запрос для подсчёта количества работодателей, у которых не указана сфера деятельности. 

In [311]:
# запрос для подсчета количества работодателей без сфер деятельности
query_5_4 = f'''select count(e.name) cnt_employers
from public.employers e
left join public.employers_industries ei on ei.employer_id = e.id
where ei.industry_id is null
'''

In [312]:
# результат запроса
df_5_4 = pd.read_sql_query(query_5_4, connection)
df_5_4

cnt_employers
0           8419

5. Напишите запрос, чтобы узнать название компании, находящейся на третьем месте в алфавитном списке (по названию) компаний, у которых указано четыре сферы деятельности. 

In [313]:
# вычисление работодателя №3, у которого 4 сферы деятельности
query_5_5 = f'''select e.name name_employer
from public.employers e
join public.employers_industries ei on ei.employer_id = e.id
group by e.name
having count(ei.industry_id) = 4
order by e.name
offset 2 limit 1
'''

In [314]:
# результат запроса
df_5_5 = pd.read_sql_query(query_5_5, connection)
df_5_5

name_employer
0          2ГИС

6. С помощью запроса выясните, у какого количества работодателей в качестве сферы деятельности указана Разработка программного обеспечения.


In [315]:
# запрос на вычисление количества работодателей с определенной сферой деятельности
query_5_6 = f'''select count(e.*) cnt_employer
from public.employers_industries ei
join public.employers e on ei.employer_id = e.id
join public.industries i on ei.industry_id = i.id
where i.name ilike '%разработка%программного%обеспечения%'
'''

In [316]:
# результат запроса
df_5_6 = pd.read_sql_query(query_5_6, connection)
df_5_6

cnt_employer
0          3553

7. Для компании «Яндекс» выведите список регионов-миллионников, в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавьте строку Total с общим количеством вакансий компании. Результат отсортируйте по возрастанию количества.

Список городов-милионников надо взять [отсюда](https://ru.wikipedia.org/wiki/%D0%93%D0%BE%D1%80%D0%BE%D0%B4%D0%B0-%D0%BC%D0%B8%D0%BB%D0%BB%D0%B8%D0%BE%D0%BD%D0%B5%D1%80%D1%8B_%D0%A0%D0%BE%D1%81%D1%81%D0%B8%D0%B8). 

Если возникнут трудности с этим задание посмотрите материалы модуля  PYTHON-17. Как получать данные из веб-источников и API. 

In [317]:
# код для получения списка городов-милионников
import urllib.parse as req

path_common = req.urlparse('https://ru.wikipedia.org/wiki/Города-миллионеры_России#Список_городов-миллионеров')
url = path_common.scheme + '://' + req.quote(path_common.netloc) +  req.quote(path_common.path) + '#'+ req.quote(path_common.fragment)

table = pd.read_html(url)[1]
df = table['Город'].replace('Воронеж[a]','Воронеж')
cities = tuple(df)
cities

('Москва',
 'Санкт-Петербург',
 'Новосибирск',
 'Екатеринбург',
 'Казань',
 'Нижний Новгород',
 'Челябинск',
 'Красноярск',
 'Самара',
 'Уфа',
 'Ростов-на-Дону',
 'Омск',
 'Краснодар',
 'Воронеж',
 'Пермь',
 'Волгоград')

In [318]:
# код на выведние городов-миллионников с количеством вакансий
query_5_7 = f'''select a.name, count(v.*)
from public.vacancies v
join public.areas a on a.id = v.area_id
join public.employers e on e.id = v.employer_id
where e.name = 'Яндекс' 
and a.name in('Москва', 'Санкт-Петербург', 'Новосибирск', 'Екатеринбург', 'Казань', 'Нижний Новгород', 
'Челябинск', 'Красноярск', 'Самара', 'Уфа', 'Ростов-на-Дону', 'Омск', 'Краснодар', 'Воронеж','Пермь',
'Волгоград')
group by a.name

union all
select 'Total', count(v.*) 
from public.vacancies v
join public.areas a on a.id = v.area_id
join public.employers e on e.id = v.employer_id
where e.name = 'Яндекс'
and a.name in('Москва', 'Санкт-Петербург', 'Новосибирск', 'Екатеринбург', 'Казань', 'Нижний Новгород', 
'Челябинск', 'Красноярск', 'Самара', 'Уфа', 'Ростов-на-Дону', 'Омск', 'Краснодар', 'Воронеж','Пермь',
'Волгоград')
order by 2 asc
'''
# в конце выводим общее количество вакансий

In [319]:
# результат запроса
df_5_7 = pd.read_sql_query(query_5_7, connection)
df_5_7

name  count
0              Омск     21
1         Челябинск     22
2        Красноярск     23
3         Волгоград     24
4             Пермь     25
5            Казань     25
6    Ростов-на-Дону     25
7               Уфа     26
8            Самара     26
9         Краснодар     30
10          Воронеж     32
11      Новосибирск     35
12  Нижний Новгород     36
13     Екатеринбург     39
14  Санкт-Петербург     42
15           Москва     54
16            Total    485

***

*Выводы:*
- "Яндекс" предлагает наибольшее количество вакансий в наибольших количествах регионах, из них среди городов-миллионников 485 вакансий, что составляет четверть всех вакансий компании. Количество вакансий растет пропорционально росту количества населения. 
- на втором месте "Ростелеком", на третьем "Спецремонт"
- количество работодателей без указания сферы деятельности - 8419, что составляет почти треть всех вакансий в базе
- работодателей со сферой деятельности Разработка программного обеспечения составляет 3553, что составляет почти седьмую часть всех работодателей, что говорит о популярности таких компании.

# Юнит 6. Предметный анализ

1. Сколько вакансий имеет отношение к данным?

Считаем, что вакансия имеет отношение к данным, если в её названии содержатся слова 'data' или 'данн'.

*Подсказка: Обратите внимание, что названия вакансий могут быть написаны в любом регистре.* 


In [320]:
# код для получения количества вакансий, относящихся к данным
query_6_1 = f'''select count(*) cnt_data
from vacancies
where name ilike '%data%' or name ilike'%данн%'
'''

In [321]:
# результат запроса
df_6_1 = pd.read_sql_query(query_6_1, connection)
df_6_1

cnt_data
0      1771

2. Сколько есть подходящих вакансий для начинающего дата-сайентиста? 
Будем считать вакансиями для дата-сайентистов такие, в названии которых есть хотя бы одно из следующих сочетаний:
* 'data scientist'
* 'data science'
* 'исследователь данных'
* 'ML' (здесь не нужно брать вакансии по HTML)
* 'machine learning'
* 'машинн%обучен%'

** В следующих заданиях мы продолжим работать с вакансиями по этому условию.*

Считаем вакансиями для специалистов уровня Junior следующие:
* в названии есть слово 'junior' *или*
* требуемый опыт — Нет опыта *или*
* тип трудоустройства — Стажировка.
 

In [322]:
# код для получения количества вакансий датасайентистов junior
query_6_2 = f'''select count(*) cnt_DS_jun
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
 	and
 	(name ilike '%junior%'
  	or experience = 'Нет опыта'
  	or employment='Стажировка')
   '''

In [323]:
# результат запроса
df_6_2 = pd.read_sql_query(query_6_2, connection)
df_6_2

cnt_ds_jun
0          51

3. Сколько есть вакансий для DS, в которых в качестве ключевого навыка указан SQL или postgres?

** Критерии для отнесения вакансии к DS указаны в предыдущем задании.*

In [324]:
# код для получения количества вакансий DS с ключевыми навыками SQL или postgres 
query_6_3 = f'''select count(*) cnt_DS_sql
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
 	and
	(key_skills ilike '%postgres%'
 or key_skills ilike '%SQL%')'''

In [325]:
# результат запроса
df_6_3 = pd.read_sql_query(query_6_3, connection)
df_6_3

cnt_ds_sql
0         201

4. Проверьте, насколько популярен Python в требованиях работодателей к DS.Для этого вычислите количество вакансий, в которых в качестве ключевого навыка указан Python.

** Это можно сделать помощью запроса, аналогичного предыдущему.*

In [326]:
# код для получения количества вакансий DS с ключевыми навыками Python
query_6_4 = f'''select count(*) cnt_DS_python
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
 	and
	(key_skills ilike '%python%')
 '''


In [327]:
# результат запроса
df_6_4 = pd.read_sql_query(query_6_4, connection)
df_6_4

cnt_ds_python
0            351

5. Сколько ключевых навыков в среднем указывают в вакансиях для DS?
Ответ округлите до двух знаков после точки-разделителя.

In [328]:
# код на вычисления среднего количества ключевых навыков в вакансиях DS
query_6_5 = f'''select round(avg(length(key_skills) - length(replace(key_skills, CHR(9), ''))+1), 2) as avg_ind_DS
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
'''

In [329]:
# результат запроса
df_6_5 = pd.read_sql_query(query_6_5, connection)
df_6_5

avg_ind_ds
0        6.41

6. Напишите запрос, позволяющий вычислить, какую зарплату для DS в **среднем** указывают для каждого типа требуемого опыта (уникальное значение из поля *experience*). 

При решении задачи примите во внимание следующее:
1. Рассматриваем только вакансии, у которых заполнено хотя бы одно из двух полей с зарплатой.
2. Если заполнены оба поля с зарплатой, то считаем зарплату по каждой вакансии как сумму двух полей, делённую на 2. Если заполнено только одно из полей, то его и считаем зарплатой по вакансии.
3. Если в расчётах участвует null, в результате он тоже даст null (посмотрите, что возвращает запрос select 1 + null). Чтобы избежать этой ситуацию, мы воспользуемся функцией [coalesce](https://postgrespro.ru/docs/postgresql/9.5/functions-conditional#functions-coalesce-nvl-ifnull), которая заменит null на значение, которое мы передадим. Например, посмотрите, что возвращает запрос `select 1 + coalesce(null, 0)`

Выясните, на какую зарплату в среднем может рассчитывать дата-сайентист с опытом работы от 3 до 6 лет. Результат округлите до целого числа. 

In [330]:
# код для вычисления уровня средней з/п может рассчитывать DS с опытом работы от 3 до 6 лет
query_6_6 = f'''select round(avg(coalesce((salary_from+salary_to)/2, salary_from, salary_to)), 0) salary_DS
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
    and (experience = 'От 3 до 6 лет')
'''

In [331]:
# результат запроса
df_6_6 = pd.read_sql_query(query_6_6, connection)
df_6_6

salary_ds
0   243115.0

***

*Выводы:*
- вакансий, связанных с данными, составляет 1771, а датасайентистов составляет 480
- из вакансий DS начинающих специалистов набирающих 1/9 от всез вакансий DS
- с навыком Python-351, что составляет 3/4 всех вакансий DS, c навыком SQL -201, что составляет почти половину вакансий DS
- среднее количество ключевых навыков указываемых в вакансиях DS составляет 6.41
- средняя указываемая зарплата DS с опытом работы от 3 до 6 лет составляет 243115 руб, что говорит о востребованности и высокоплачеваемости труда.

# Общий вывод по проекту

In [332]:
# код для вычисления количества вакансий Data Science в базе
query_7_1 = f'''select count(*) cnt_ds
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
'''

In [333]:
df_7_1 = pd.read_sql_query(query_7_1, connection)
df_7_1

cnt_ds
0     480

In [334]:
# код для вычисления уровня средней з/п может рассчитывать DS с опытом работы от 1 до 3 лет
query_7_2 = f'''select round(avg(coalesce((salary_from+salary_to)/2, salary_from, salary_to)), 0) salary_DS_1_3
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
    and (experience = 'От 1 года до 3 лет')
'''
# код для вычисления уровня средней з/п может рассчитывать DS без опыта работы
query_7_3 = f'''select round(avg(coalesce((salary_from+salary_to)/2, salary_from, salary_to)), 0) salary_DS_junior
from vacancies
where (name ilike '%data scientist%'
   	or name ilike '%data science%'
   	or name ilike '%исследователь данных%'
   	or (name like '%ML%' and name not ilike '%HTML%')
   	or name ilike '%machine learning%' or name ilike '%машинн%обучен%')
    and (experience = 'Нет опыта')
'''

In [335]:
df_7_2 = pd.read_sql_query(query_7_2, connection)
df_7_2 # средняя рассчитываемая з/п для DS c опытом от 1 до 3 лет

salary_ds_1_3
0       139675.0

In [336]:
df_7_3 = pd.read_sql_query(query_7_3, connection)
df_7_3 # средняя рассчитываемая з/п для DS без опыта работы

salary_ds_junior
0           74643.0

In [348]:
# код для получения топ-10 списка компаний, которые предоставляет вакансии DS 
query_7_4 = f'''select e.name employer, count(v.*) cnt_vac
from public.vacancies v 
join public.employers e on v.employer_id = e.id
where (v.name ilike '%data scientist%'
   	or v.name ilike '%data science%'
   	or v.name ilike '%исследователь данных%'
   	or (v.name like '%ML%' and v.name not ilike '%HTML%')
   	or v.name ilike '%machine learning%' or v.name ilike '%машинн%обучен%')
group by e.id
order by 2 desc
limit 10
'''

In [349]:
df_7_4 = pd.read_sql_query(query_7_4, connection)
df_7_4

employer  cnt_vac
0                   СБЕР       37
1        Bell Integrator       25
2         Банк ВТБ (ПАО)       18
3                     VK       15
4  Positive Technologies       11
5         EvenBet Gaming        9
6                 Яндекс        9
7                МегаФон        8
8               Andersen        7
9                    МТС        6

*Итоги:*
- вакансии, связанные с данными = 1771, что составляет 3.59% всех вакансий, а вакансий Data Scientist = 480, что составляет 0.97% всех вакансий в базе
- Датасайентист может рассчитывать на размер зарплаты, пропорционально опыту работы. Начинающий DS в среднем на 74643 руб, c опытом работы от 1 до 3 лет = 139675 руб, а с опытом от 3 до 6 лет = 243115руб. Но это примерные данные, размер оплаты зависит от многих факторов, не только от продолжительности опыта работы.
- можно исследовать какие компании предлагают наибольшие средние зарплаты в зависимости от опыта работы, региона предоставления вакансий. Например: в каких регионах самая большая зарплата предлагается.
- компания, которая набирает наибольшее количество датасайентистов - СБЕР.
- также можно исследовать какие работодатели, с какой сферой деятельности, набирают в штат датасайентистов.